### ISSUE TITLE
RFE: Listen to side menu button clicked event

### ISSUE THREAD

**Issue by 3lix** [codenameone/CodenameOne#1907]
Per this stackoverflow conversation with Chen : http://stackoverflow.com/questions/39909501/listen-to-side-menu-hamburger-menu-clicked-event

I would like to listen to side menu clicked event for the purpose to further customize the side menu. Mainly I am interested in 
1) Highlighting the already selected command option when the side menu opens
2) Tinting the form that is no longer in focus because of the side menu as a visual clue

**Comment by codenameone** [comment]
What about swiping the menu in?

**Comment by 3lix** [comment]
Can you please elaborate. I am not sure I understand the  question.

**Comment by codenameone** [comment]
Side menus can be open by swiping the screen from left to right.

**Comment by 3lix** [comment]
Yes I hear your concern.  Honestly in my app I may take the route of disabling swiping for the side menu.  

To my very specific use case, with disabling swiping, I gain a lot from being able to listen to the hamburger menu clicked event. I understand that other users may not want to take this trade off in their own app. 

I feel, if it is fairly simple to expose the button clicked event in the API, we should do it. This has been the spirit of Codename one where the platform empowers developers to use it in a new/creative way. I also accept the trade off of losing swiping since I value a lot more showing the previously selected command  in addition to tinting.   

If you feel strongly against exposing such call, I also understand. Maybe this enhancement could be again considered in the future. 
Thanks.
